File: p7/p7.c
#include <stdio.h>
#include <stdbool.h>
#include <math.h>
/*
 * returns true iff n is a prime else false
 */
bool isPrime(int n) {
	for(int i = 2; i <= sqrt(n); i++) {
		if(n % i == 0) {
			return false;
		}
	}
	return true;
}

/*
 * returns the nth prime
 */
int prime(int n) {
	if(n < 3) {
		return (n == 1)? 2 : 3;
	} else {
		int p = 2, nthPrime;
		for(int i = 4; p < n; i++) {
			if(isPrime(i)) {
				nthPrime = i;
				p++;
			}
		}
		return nthPrime;
	}
}

int main() {
	printf("%d\n", prime(10001));
	return 0;
}

File: P7/P7.java
/* Problem 7
 *
 * By listing the first six prime numbers: 2, 3, 5, 7, 11, and 13,
 * we can see that the 6th prime is 13.
 *
 * What is the 10 001st prime number?
 */

final class P7 {
	static boolean isPrime(int n) {
		int root = (int) Math.sqrt(n);
		for (int i = 2; i <= root; i++) {
			if (n % i == 0) return false;
		}
		return true;
	}

	public static void main(String[] args) {
		int nth_prime = 10001;
		int p = 1;

		do {
			p++;
			if (isPrime(p)) nth_prime--;
		} while (nth_prime > 0);

		System.out.println(p);
	}
}
